src/MiniProfiler.AspNetCore/MiniProfilerExtensions.cs

```diff
@@ -40,33 +53,76 @@ public static class MiniProfilerExtensions
-            var authorized = state?.IsAuthorized ?? false;
-            var ids = state?.RequestIDs ?? Enumerable.Empty<Guid>();
+            sb.Append("<script async id=\"mini-profiler\" src=\"");
+            sb.Append(path);
+            sb.Append("includes.js?v=");
+            sb.Append(version);
+            sb.Append("\" data-version=\"");
+            sb.Append(version);
+            sb.Append("\" data-path=\"");
+            sb.Append(path);
+            sb.Append("\" data-current-id=\"");
+            sb.Append(profiler.Id.ToString());
 
-            if (!MiniProfilerMiddleware.Current.Embedded.TryGetResource("include.partial.html", out string format))
+            sb.Append("\" data-ids=\"");
+            var ids = state?.RequestIDs;
+            if (ids != null)
             {
-                return includeNotFound;
+                var first = true;
+                var length = ids.Count;
+                for (var i = 0; i < length; i++)
+                {
+                    if (first)
+                    {
+                        first = false;
+                    }
+                    else
+                    {
+                        sb.Append(',');
+                    }
+                    var id = ids[i];
+                    sb.Append(id.ToString());
+                }
             }
 
-            Func<bool, string> toJs = b => b ? "true" : "false";
-
-            var sb = new StringBuilder(format);
-            sb.Replace("{path}", MiniProfilerMiddleware.Current.BasePath.Value.EnsureTrailingSlash())
-              .Replace("{version}", MiniProfiler.Settings.VersionHash)
-              .Replace("{currentId}", profiler.Id.ToString())
-              .Replace("{ids}", string.Join(",", ids.Select(guid => guid.ToString())))
-              .Replace("{position}", (position ?? MiniProfiler.Settings.PopupRenderPosition).ToString().ToLower())
-              .Replace("{showTrivial}", toJs(showTrivial ?? MiniProfiler.Settings.PopupShowTrivial))
-              .Replace("{showChildren}", toJs(showTimeWithChildren ?? MiniProfiler.Settings.PopupShowTimeWithChildren))
-              .Replace("{maxTracesToShow}", (maxTracesToShow ?? MiniProfiler.Settings.PopupMaxTracesToShow).ToString())
-              .Replace("{showControls}", toJs(showControls ?? MiniProfiler.Settings.ShowControls))
-              .Replace("{authorized}", toJs(authorized))
-              .Replace("{toggleShortcut}", MiniProfiler.Settings.PopupToggleKeyboardShortcut)
-              .Replace("{startHidden}", toJs(startHidden ?? MiniProfiler.Settings.PopupStartHidden))
-              .Replace("{trivialMilliseconds}", MiniProfiler.Settings.TrivialDurationThresholdMilliseconds.ToString());
-            return new HtmlString(sb.ToString());
+            sb.Append("\" data-position=\"");
+            sb.Append((position ?? MiniProfiler.Settings.PopupRenderPosition).ToString().ToLower());
```

Comment:
After this commit I have an urge to `switch` that `.ToString().ToLower()`. Damn you, instigator.